experiments/gym/train_dqn_cartpole.py

```diff
@@ -0,0 +1,304 @@
+    args,
+    logged_cbs: list[LoggingMixin],
+    cbs_name_to_checkpoint: list[str] = None,
+):
+    """The function creates the supplementary callbacks for the training and adds them to the callback lists
+    and returns the list.
+
+        Arguments:
+            args: input args
+            logged_cbs (list[Callback]): the list of callbacks
+            cbs_name_to_checkpoint (list[str]): the name of callbacks to checkpoint
+        Returns:
+            (list[Callback]): the full list of callbacks for the training
+    """
+    if args.use_wandb:
+        from emote.callbacks.wb_logger import WBLogger
+
+        config = {
+            "wandb_project": args.name,
+            "wandb_run": args.wandb_run,
+            "hidden_dims": args.hidden_layer_size,
+            "batch_size": args.batch_size,
+            "learning_rate": args.actor_lr,
+            "rollout_len": args.rollout_length,
+        }
+        logger = WBLogger(
+            callbacks=logged_cbs,
+            config=config,
+            log_interval=100,
+        )
+    else:
+        logger = TensorboardLogger(
+            logged_cbs,
+            SummaryWriter(log_dir=args.log_dir + "/" + args.name + "_{}".format(time.time())),
+            100,
+        )
+
+    bp_step_terminator = BackPropStepsTerminator(bp_steps=args.bp_steps)
+    callbacks = logged_cbs + [logger, bp_step_terminator]
+
+    if cbs_name_to_checkpoint:
+        checkpointer = Checkpointer(
+            callbacks=[
+                cb for cb in logged_cbs if hasattr(cb, "name") and cb.name in cbs_name_to_checkpoint
+            ],
+            run_root=args.checkpoint_dir,
+            checkpoint_interval=args.checkpoint_interval,
+        )
+        callbacks += [checkpointer]
+
+    return callbacks
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--name", type=str, default="cartpole")
+    parser.add_argument("--log-dir", type=str, default="./mllogs/emote/cartpole")
+    parser.add_argument("--num-envs", type=int, default=4)
+    parser.add_argument("--rollout-length", type=int, default=1)
+    parser.add_argument("--batch-size", type=int, default=128)
+    parser.add_argument("--hidden-dims", type=list, default=[128, 128])
+    parser.add_argument("--lr", type=float, default=1e-3, help="The learning rate")
+    parser.add_argument("--device", type=str, default="cpu")
+    parser.add_argument("--bp-steps", type=int, default=50_000)
+    parser.add_argument("--memory-size", type=int, default=50_000)
+    parser.add_argument("--export-memory", action="store_true", default=False)
+    parser.add_argument("--use-wandb", action="store_true")
+    parser.add_argument(
+        "--wandb-run",
+        type=str,
+        default=None,
+        help="Short display name of run for the W&B UI. Randomly generated by default.",
+    )
+    arg = parser.parse_args()
+
+    env = DictGymWrapper(AsyncVectorEnv([_make_env() for _ in range(arg.num_envs)]))
```

Comment:
Could we break this out into a `main(args)`? I think that tends to look a bit cleaner.